geonode_mapstore_client/client/js/epics/gnsearch.js

```diff
@@ -171,10 +171,10 @@ const requestResourcesObservable = ({
 // checks if location change is made to a viewer page
 const isViewerPage = (currentPath) => {
     if (currentPath === '/') return false;
-    const resourcePath = currentPath.split('/')[1];
-    const resourceStrings = ['dataset', 'map', 'dashboard', 'geostory'];
-    const pathMatchResource = resourceStrings.some(res => res.match(resourcePath));
-    return pathMatchResource;
+    const match = CATALOGUE_ROUTES.filter(route => route.component === appRouteComponentTypes.resourceViewer).some(route => {
```

Comment:
We cannot rely on `route.component` because we need to include also the upload pages for this reason I propose to add a new property to the route called shouldNotRequestResources and we can add this only to the page that need it. Finally the filter will become `.filter(route => route.shouldNotRequestResources)`